src/protagonist/Engine/Ingest/Image/ImageServer/Clients/CantaloupeThumbsClient.cs

```diff
@@ -84,6 +93,37 @@ public class CantaloupeThumbsClient : IThumbsClient
         throw new HttpException(response.StatusCode, "failed to retrieve data from the thumbs processor");
     }
 
+    private HttpRequestMessage CreateCantaloupeRequestMessage(string convertedS3Location, string size)
+    {
+        var request = new HttpRequestMessage(HttpMethod.Get, $"iiif/3/{convertedS3Location}/full/{size}/0/default.jpg");
+
+        if (loadBalancerCookies.Any())
+        {
+            request.Headers.Add("Cookie", loadBalancerCookies);
+        }
+
+        return request;
+    }
+
+    private void AttemptToAddStickinessCookie(HttpResponseMessage response)
+    {
+        if (!cantaloupeClient.DefaultRequestHeaders.Contains("Cookie"))
+        {
+            var hasCookie = response.Headers.TryGetValues("Set-Cookie", out var cookies);
+            if (hasCookie)
+            {
+                foreach (var cookie in cookies!)
+                {
+                    if (CookieHeaderValue.TryParse(cookie, out var parsedCookie) && parsedCookie.Cookies.Any(x =>
+                            engineSettings.ImageIngest!.LoadBalancerStickinessCookieNames.Contains(x.Name)))
+                    {
+                        loadBalancerCookies.Add(cookie);
```

Comment:
Does this need to be a dict? Can cookies with same name but differing values or expires be returned? How are they handled if sent to ALB?

Reply:
the `cookie` being added is actually a string containing all the values of the cookie, so it would have the correct expiry and values for the load balancer - thought this was ok, as this is only set for the lifecycle of a request, and every new request would get these values added - should note the `Name` value taken from the parsed cookie is the key of a kvp that doesn't change (though I left it as configurable via app settings)